sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala

```diff
@@ -124,6 +104,24 @@ case class CustomShuffleReaderExec private(
     Map("numSkewedPartitions" -> metrics)
   }
 
+  private def sendPartitionDataSizeMetrics(executionId: String): Unit = {
+    val mapStats = shuffleStage.get.mapStats.bytesByPartitionId
+    partitionSpecs.foreach {
+      case CoalescedPartitionSpec(startReducerIndex, endReducerIndex) =>
+        val dataSize = startReducerIndex.until(endReducerIndex).map(mapStats(_)).sum
+        metrics("partitionDataSize").set(dataSize)
+        SQLMetrics.postDriverMetricUpdates(
+          sparkContext, executionId,
+          metrics.filter(_._1 == "partitionDataSize").values.toSeq)
```

Comment:
can we look up the `partitionDataSize` `SQLMetric` at the beginning of this method? then here we can simply write `Seq(metric)`.